boards/arm/hexiwear_k64/Kconfig.defconfig

```diff
@@ -49,25 +49,6 @@ config PINMUX_MCUX_PORTE
 
 endif # PINMUX_MCUX
 
-if GPIO_MCUX
-
-config GPIO_MCUX_PORTA
-	default y if MAX30101
```

Comment:
we seem to be losing this conditional enable

Reply:
We are, it's a bit of the tradeoff to have DT being the singular source of how the HW is configured, we don't have a great means beyond overlays to reclaim this functionality.

We could look at an optimization to only generate these instances if there is some reference to them.

Reply:
This tradeoff has an impact to power consumption and to a lesser degree memory footprint, so we need to figure something out.

Reply:
We can still turn off individual ports, via device tree.

I updated things so GPIO PORTA is disabled by default and enabled in the MAX30101 sample.